### ISSUE TITLE
Support GitLab System Hooks

### ISSUE THREAD

**Issue by JadeQin** [jenkinsci/gitlab-plugin#836]
### Context
- **Gitlab plugin version**: 1.5.9
- **Gitlab version**: 10.7.3
- **Jenkins version**: 2.73.3
- **Job type**: Pipeline

### Logs & Traces
```
<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8"/>
<title>Error 404 Not Found</title>
</head>
<body><h2>HTTP ERROR 404</h2>
<p>Problem accessing /project/CI-Code-Analysis. Reason:
<pre>    Not Found</pre></p><hr><a href="http://eclipse.org/jetty">Powered by Jetty:// 9.4.z-SNAPSHOT</a><hr/>

</body>
</html>
```

### Problem description
I add a global hook in gitlab ,but it return 404 when it trigger,but it works good when I add it in single project.I check the param,when it is global ,the head params like this:
```
Content-Type: application/json
X-Gitlab-Event: System Hook
```
But when it in project,the head params like this:
```
Content-Type: application/json
X-Gitlab-Event: Merge Request Hook
```

I look the code maybe about it at https://github.com/jenkinsci/gitlab-plugin/blob/518063be653de672c574cc86536c7617fb0e5159/src/main/java/com/dabsquared/gitlabjenkins/webhook/ActionResolver.java#L107
```
switch (eventHeader) {
            case "Merge Request Hook":
                return new MergeRequestBuildAction(project, getRequestBody(request), tokenHeader);
            case "Push Hook":
            case "Tag Push Hook":
                return new PushBuildAction(project, getRequestBody(request), tokenHeader);
            case "Note Hook":
                return new NoteBuildAction(project, getRequestBody(request), tokenHeader);
            case "Pipeline Hook":
                return new PipelineBuildAction(project, getRequestBody(request), tokenHeader);
            default:
                LOGGER.log(Level.FINE, "Unsupported X-Gitlab-Event header: {0}", eventHeader);
                return new NoopAction();
        }
```
So,the plugin do not handle the `System Hook` head.

How can I resolve the problem?Thanks

**Comment by omehegan** [comment]
The plugin does not currently support GitLab System Hooks. https://docs.gitlab.com/ee/system_hooks/system_hooks.html

**Comment by omehegan** [comment]
This was released today in 1.5.10.
